### PR TITLE
Asgardeo api authentication

### PR DESCRIPTION
**Pull request by himeshsiriwardana**
https://github.com/wso2-enterprise/asgardeo-product/issues/25174

Updated both management API auth and organization API auth documentation.

Organization API auth doc:

screencapture-localhost-8000-en-7-0-0-apis-organization-apis-authentication-2024-06-28-15_09_16

### CODE REVIEW

**Comment by sadilchamishka** [comment]
For a created user -> For a user managed in the organization
For an invited user -> For a user invited to the organization from the parent organization.

Having detail topic might be helpful to understand.

File: en/identity-server/7.0.0/docs/apis/organization-apis/authentication.md [hunk 1]
```diff
@@ -81,35 +111,52 @@ https://{{ host_name }}/oauth2/authorize?response_type=code&redirect_uri={redire
...
+After receiving the authorization code, the application has to exchange it to get the `id_token` and the `access_token`. The method of obtaining these tokens differs depending on the type of user which can be one of the following.
 
-### Step 1: For the organization (root)
-After receiving the authorization code, the application has to exchange it to get the tokens given below:
+  - A user that was created within the organization.
+  - A parent organization user that was invited to join the organization.
 
-- `access_token`
-- `id_token`
+### For a created user
```
Comment (AnuradhaSK) [inline en/identity-server/7.0.0/docs/apis/organization-apis/authentication.md:131]: Better to reword this
+1 for https://github.com/wso2/docs-is/pull/4636#issuecomment-2202287279 suggestion 

Update in both doc spaces

File: en/identity-server/7.0.0/docs/apis/organization-apis/authentication.md [hunk 2]
```diff
@@ -1,72 +1,102 @@
...
+    - Learn more about API authorization in [API authorization with RBAC]({{base_path}}/guides/api-authorization/).
 
-## Get the authorization code
-First, your application must initiate a login request to the authorization endpoint of {{ product_name }}. After redirecting to {{ product_name }}, the user should be prompted with a login page if the user is not authenticated.
 
-**Request Format**
-``` bash
-https://{{ host_name }}/oauth2/authorize?response_type=code&redirect_uri={redirect_uri}&client_id={client_id}
-```
+## Step 4: Get the authorization code
```
Comment (AnuradhaSK) [inline en/identity-server/7.0.0/docs/apis/organization-apis/authentication.md:78]: Before this step 

need to mention that, roles associated to the application will be available in the application shared organizations.
At the organization level, the roles need to be assigned to the user

Comment (himeshsiriwardana) [reply en/identity-server/7.0.0/docs/apis/organization-apis/authentication.md]: Added the steps specifically to complete the flow. Can you review it? @AnuradhaSK 

screencapture-localhost-8000-asgardeo-docs-apis-organization-apis-authentication-2024-07-08-18_21_15

**Review by sadilchamishka — APPROVED** [review]

**Pull request merged**